From patchwork Mon Feb 24 08:13:57 2025
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Neha Malcom Francis <n-francis@ti.com>
X-Patchwork-Id: 24896
X-Patchwork-Delegate: praneeth@ti.com
From: Neha Malcom Francis <n-francis@ti.com>
Subject: [tiU25.01 PATCH] TI: HACK: arm: dts: k3-j784s4-evm-u-boot: Add
 PBIST_14 node
Date: Mon, 24 Feb 2025 13:43:57 +0530
Message-ID: <20250224081357.1736052-1-n-francis@ti.com>
Return-Path: nm@ti.com
List-Post: <mailto:linux-patch-review@list.ti.com>
MIME-Version: 1.0
To: u-kumar1@ti.com, praneeth@ti.com
Cc: b-padhi@ti.com, m-chawdhry@ti.com, n-francis@ti.com,
 linux-patch-review@list.ti.com
List-ID: <linux-patch-review@list.ti.com>
List-Post: <mailto:linux-patch-review@list.ti.com>

Add DT node for PBIST_14 that is responsible for triggering the PBIST
self-tests for the MAIN_R5_2_x cores. The patch (link) upstreaming this
node to its correct location in dts/upstream via kernel is still under
discussion so placing it here temporarily.

Link: https://lore.kernel.org/all/20241128140825.263216-1-n-francis@ti.com/
Signed-off-by: Neha Malcom Francis <n-francis@ti.com>
---
 arch/arm/dts/k3-j784s4-evm-u-boot.dtsi | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm/dts/k3-j784s4-evm-u-boot.dtsi b/arch/arm/dts/k3-j784s4-evm-u-boot.dtsi
index f3a09887975..fbfd753be76 100644
--- a/arch/arm/dts/k3-j784s4-evm-u-boot.dtsi
+++ b/arch/arm/dts/k3-j784s4-evm-u-boot.dtsi
@@ -163,3 +163,17 @@
 &usb0 {
 	dr_mode = "peripheral";
 };
+
+&cbass_main {
+	bist_main14: bist@33c0000 {
+		compatible = "ti,j784s4-bist";
+		reg-names = "cfg", "ctrl_mmr";
+		reg = <0x00 0x033c0000 0x00 0x400>,
+			  <0x00 0x0010c1a0 0x00 0x01c>;
+		clocks = <&k3_clks 237 7>;
+		power-domains = <&k3_pds 237 TI_SCI_PD_EXCLUSIVE>;
+		bootph-pre-ram;
+		ti,bist-instance = <14>;
+		cores-under-test = <&main_r5fss2_core0>, <&main_r5fss2_core1>;
+	};
+};
